qdrant-landing/content/documentation/concepts/points.md

```diff
@@ -393,15 +393,65 @@ client
     .await?;
 ```
 
-<!-- 
 
```

Comment:
Basic questions:

- by "everything in the memory", I'm guessing that you're referring to RAM
- I don't see any `upload_collection` or `upload_points` methods in our docs or API (if we have them, I'd want to link to them)
- Do we have a Qdrant `upsert API`, or is this a generic version?
- I see parallelization and retry support in your code samples. But I don't see an example of code that enables lazy batching support

Reply:
> by "everything in the memory", I'm guessing that you're referring to RAM
yes

> I don't see any upload_collection or upload_points methods in our docs or API (if we have them, I'd want to link to them)

These are python-only methods provided for the convenience (they are wrappers around basic functionality)

> Do we have a Qdrant upsert API, or is this a generic version?

We sometimes refer to upload points as upsert
https://qdrant.tech/documentation/concepts/points/#upload-points

 > I see parallelization and retry support in your code samples. But I don't see an example of code that enables lazy batching support

There are lazy structures in python, e.g. generators ([example article](https://realpython.com/introduction-to-python-generators/), they do not load everything in memory from the very beginning, but load each value on demand (e.g. compute it or read from a disk). One can use them to read large files which don't fit into RAM. 

There are a set of operations which uploads the whole structure to RAM, we don't use them. We read everything iteratively, construct a batch and send it. So the number of points which are loaded into RAM simultaneously is limited and users can upload large files without an effort.